### PR TITLE
feat: add JSDoc to parser

### LINKED ISSUE #123: Add JSDoc to usage-parser
Please, add [JSDoc](http://usejsdoc.org/) to usage-parser.

### PR DESCRIPTION
**Pull request by ifedchankau**
Add JSDoc to parser

Closes #123

### CODE REVIEW

File: src/getContext.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
 'use strict';
 
+// Import templates
 const argumentTemplate = require('./template/argument.json');
 const context = require('./template/context.js');
```
Comment (romanrostislavovich) [inline src/getContext.js]: I think `context` it's not template

File: src/configValidation.js [hunk 2]
```diff
@@ -17,6 +30,11 @@ const schemaValidating = (config) => {
     }
 };
 
+/**
+ * Fill empty configuration fields with defaults
```
Comment (romanrostislavovich) [inline src/configValidation.js]: This function fill by default only missing properties in configuration

File: src/configValidation.js [hunk 3]
```diff
@@ -1,13 +1,26 @@
+// Import validation schema and default config
```
Comment (romanrostislavovich) [inline src/configValidation.js]: I think best name for that is `Import templates`

File: src/getContext.js [hunk 4]
```diff
@@ -79,9 +116,19 @@ const setSectionWithPostfixArray =
     return data;
 };
 
+/**
+ * Searches for section by name + postfix and adds it to data
```
Comment (romanrostislavovich) [inline src/getContext.js]: This description of function is duplicate?

File: src/parseSection.js [hunk 5]
```diff
@@ -27,15 +34,27 @@ const options = (section, context, argumentTemplate) => {
     }
 };
 
+/**
+ * Parse section Usage and add argument to context
```
Comment (romanrostislavovich) [inline src/parseSection.js]: Which argument(s) added to context?

File: src/parser.js [hunk 6]
```diff
@@ -1,16 +1,29 @@
 'use strict';
 
+// Import functions
 const getContext = require('./getContext.js');
 const templatizer = require('./templatizer.js');
 const validate = require('./configValidation.js');
+
+// Import default user config
```
Comment (romanrostislavovich) [inline src/parser.js]: `Import templates` ?

File: src/template/context.js [hunk 7]
```diff
@@ -1,5 +1,7 @@
+// Import parsing functions for each section
```
Comment (romanrostislavovich) [inline src/template/context.js]: Mb `Import function`, what do you think ?

**Review by romanrostislavovich — CHANGES_REQUESTED** [review]
Look comments, but looks good

File: src/template/context.js (hunk 7 above)
Comment (ifedchankau) [reply src/template/context.js]: @svoboda-rabstvo i fixed everything. Please, review.

**Review by romanrostislavovich — APPROVED** [review]

**Pull request merged**